app/src/main/scala/org/alephium/explorer/Main.scala

```diff
@@ -55,32 +55,70 @@ class BootUp extends StrictLogging {
   implicit val databaseConfig: DatabaseConfig[PostgresProfile] =
     DatabaseConfig.forConfig[PostgresProfile]("db", typesafeConfig)
 
-  implicit val actorSystem: ActorSystem           = ActorSystem()
-  implicit val executionContext: ExecutionContext = actorSystem.dispatcher
+  def init(): Unit =
+    config.bootMode match {
+      case mode: BootMode.Readable =>
+        initInReadableMode(mode)
 
-  private val explorer: ExplorerState = ExplorerState()
+      case BootMode.WriteOnly =>
+        initInWriteOnlyMode()
+    }
+
+  def initInReadableMode(mode: BootMode.Readable): Unit = {
+    implicit val actorSystem: ActorSystem           = ActorSystem()
+    implicit val executionContext: ExecutionContext = actorSystem.dispatcher
+
+    val explorer: ExplorerState =
+      mode match {
+        case BootMode.ReadOnly  => ExplorerState.ReadOnly()
+        case BootMode.ReadWrite => ExplorerState.ReadWrite()
+      }
 
-  def init(): Unit = {
     explorer
       .start()
       .onComplete {
         case Success(_) => ()
         case Failure(e) =>
-          logger.error(s"Fatal error during initialization: $e")
+          logger.error(s"Fatal error during initialization", e)
           actorSystem.terminate()
       }
 
     Runtime.getRuntime.addShutdownHook(new Thread(() => {
       sideEffect(Await.result(actorSystem.terminate(), 10.seconds))
     }))
+
+    CoordinatedShutdown(actorSystem).addTask(
+      CoordinatedShutdown.PhaseBeforeActorSystemTerminate,
+      "Shutdown services"
+    ) { () =>
+      for {
+        _ <- explorer.stop()
+      } yield Done
+    }
   }
 
-  CoordinatedShutdown(actorSystem).addTask(
-    CoordinatedShutdown.PhaseBeforeActorSystemTerminate,
-    "Shutdown services"
-  ) { () =>
-    for {
-      _ <- explorer.stop()
-    } yield Done
+  @SuppressWarnings(Array("org.wartremover.warts.GlobalExecutionContext"))
+  def initInWriteOnlyMode(): Unit = {
+    implicit val ec: ExecutionContext =
+      scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
we could create a task about properly configure an Execution context, I think there are way to avoid using the global one and be more fined grained about what we want, but I almost always used Akka, so I don't remember how ^^

Reply:
I think we've already got one #204. Can look at tuning stuff there but I'm beginning to realise that there might be no need for special tuning other than just in `ReadWrite` mode. But yep this can be looked into within that issue?